### ISSUE TITLE
Some invalid operands not detected

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#833]
from @Majkl578:

Invalid:
```php
echo "x" ^ 1;
echo "x" | new stdClass();
```

Valid:
```php
echo "x" ^ "y";
echo 4 ^ 5;
```

**Comment by weirdan** [comment]
> If both operands for the &, | and ^ operators are strings, then the operation will be performed on the ASCII values of the characters that make up the strings and the result will be a string.

http://php.net/manual/en/language.operators.bitwise.php

So `echo "x" ^ "y";` is not invalid.

**Comment by Majkl578** [comment]
Yup, it's a modified example from https://github.com/phpstan/phpstan/issues/1082.

**Comment by muglug** [comment]
Good points, I've updated the ticket
